src/DataType.cpp

```diff
@@ -0,0 +1,328 @@
+#include "open62541pp/DataType.h"
+
+#include <algorithm>  // copy
+#include <cassert>
+#include <cstring>
+#include <utility>  // move, swap
+
+#include "open62541pp/Config.h"
+#include "open62541pp/overloads/comparison.h"
+#include "open62541pp/types/NodeId.h"
+
+namespace opcua {
+
+[[maybe_unused]] inline static const char* emptyIfNullptr(const char* name) {
+    return name == nullptr ? "" : name;
+};
+
+static void clearMembers(UA_DataType* native) {
+    if (native != nullptr) {
+        delete[] native->members;  // NOLINT
+        native->members = nullptr;
+        native->membersSize = 0;
+    }
+}
+
+static void copyMembers(const DataTypeMember* members, size_t membersSize, UA_DataType* dst) {
+    if (dst != nullptr) {
+        dst->members = new DataTypeMember[membersSize];  // NOLINT
+        dst->membersSize = membersSize;
+        std::copy(
+            members,
+            members + membersSize,  // NOLINT
+            dst->members
+        );
+    }
+}
+
+static void copy(const UA_DataType* src, UA_DataType* dst) {
+    if (src != nullptr && dst != nullptr) {
+        clearMembers(dst);
+        *dst = *src;
+        copyMembers(dst->members, dst->membersSize, dst);
+    }
+}
+
+DataType::DataType(const UA_DataType& native) {
+    copy(&native, handle());
+}
+
+DataType::DataType(UA_DataType&& native)
+    : data_(native) {}
+
+DataType::DataType(TypeIndex typeIndex)
+    : DataType(UA_TYPES[typeIndex])  // NOLINT
+{
+    assert(typeIndex < UA_TYPES_COUNT);  // NOLINT
+}
+
+DataType::~DataType() {
+    clearMembers(handle());
+}
+
+DataType::DataType(const DataType& other) {
+    copy(other.handle(), handle());
+}
+
+DataType::DataType(DataType&& other) noexcept {
+    std::swap(data_, other.data_);
+}
+
+DataType& DataType::operator=(const DataType& other) {
+    if (this != &other) {
+        copy(other.handle(), handle());
+    }
+    return *this;
+}
+
+DataType& DataType::operator=(DataType&& other) noexcept {
+    if (this != &other) {
+        std::swap(data_, other.data_);
+    }
+    return *this;
+}
+
+// NOLINTNEXTLINE
+const char* DataType::getTypeName() const noexcept {
+#ifdef UA_ENABLE_TYPEDESCRIPTION
+    return handle()->typeName;
+#endif
+    return nullptr;
```

Comment:
Fix compile warning by changing this to:
```
#ifdef UA_ENABLE_TYPEDESCRIPTION
    return handle()->typeName;
#else
    return nullptr;
#endif
```

Reply:
Thanks @johanneskopf, just fixed it.